pkg/framework/spi/jwe/api.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jwe
+
+import "github.com/google/tink/go/keyset"
+
+// Provider interface for Crypto ctx
+type Provider interface {
+}
+
+// Creator method to create new pluggable JWE service
+type Creator func(prov Provider) (JWE, error)
+
+// JWE is an interface defining a pluggable JWE builder operations provider to Aries agents.
+// It's responsible for building JWE structures for either 1: sealing them or 2: open serialized ones. It depends on
+// on the AriesCrypto to encrypt/decrypt content and AriesKMS to get key handles needed for crypto operations.
+type JWE interface {
+	// Seal will encrypt and wrap msg for the sender key represented by skh and the recipients keys represented in rkh
+	//  into a JWE envelope then serializes it
+	// returns:
+	//		[]byte base64UrlEncoding of the serialized JWE envelope
+	//		error in case of an error
+	Seal(msg []byte, skh *keyset.Handle, rkh []*keyset.Handle) ([]byte, error)
```

Comment:
"Seal" is not an operation I would associate with JWEs as a whole. 

These two operations, Seal and Open, are behaviors of sodium's cryptobox, not of JWEs.

Reply:
Agree. I don't find Seal and Open intuitive.

Reply:
we usually `Seal` an envelope prior to sending it and `Open` it when we receive it.. just like real world mail envelopes

Reply:
This is not common terminology for JWE.

Reply:
I like really like Baha's phrasing. In that case I would suggest rename JWE -> Envelope.

But then again I believe there is intent to reuse this elsewhere in contexts where JWEs are explicit and center-class?

Reply:
The intent is to separate the common JWE implementation into own package.

(This common JWE implementation would be imported by the default provider implementations).

Reply:
as discussed today... JWE should not be an SPI... I will remove this interface and consolidate Crypto and KMS into 1 SPI